### PR TITLE
small fixes. added optuna script. remove colect env.

### PR DESCRIPTION
**Pull request by kprokofi**
- [x] Added optuna optimization for other hyper parameters (internal use only)
- [x] Added batch calculating for small datasets (crucial change for few datasets)
- [x] fix IR score extraction and metric computation for multilabel setup
- [x] minor changes

### CODE REVIEW

File: tools/main.py [hunk 1]
```diff
@@ -62,8 +62,6 @@ def main():
     sys.stdout = Logger(osp.join(cfg.data.save_dir, log_name))
 
     print('Show configuration\n{}\n'.format(cfg))
-    print('Collecting env info ...')
-    print('** System info **\n{}\n'.format(collect_env_info()))
```
Comment (sovrasov) [inline tools/main.py:66]: Why did you remove this?

File: configs/multilabel_classification/mobilenetv3_large.yml [hunk 2]
```diff
@@ -65,7 +65,7 @@ train:
   nbd: True
   max_epoch: 80
   weight_decay: 5e-4
-  batch_size: 64
+  batch_size: 8
```
Comment (sovrasov) [inline configs/multilabel_classification/mobilenetv3_large.yml]: Seems like 8 is not suitable for more or less large datasets.

File: torchreid/data/datamanager.py [hunk 3]
```diff
@@ -342,3 +342,10 @@ def __init__(
         print('  target            : {}'.format(self.targets))
         print('  *****************************************')
         print('\n')
+
+    @staticmethod
+    def calculate_batch(cur_batch, data_len):
+        ''' This heuristic is essential for training small datasets '''
```
Comment (sovrasov) [inline torchreid/data/datamanager.py]: This heuristic improves multilabel training on ...

File: torchreid/data/datamanager.py [hunk 4]
```diff
@@ -236,13 +236,13 @@ def __init__(
         assert isinstance(self._num_train_pids, list)
         assert isinstance(self._num_train_cams, list)
         assert len(self._num_train_pids) == len(self._num_train_cams)
-
+        batch_size_train = self.calculate_batch(batch_size_train, len(train_dataset))
```
Comment (sovrasov) [inline torchreid/data/datamanager.py]: I'd rather disable that by default since it is useful only for certain datasets and training configs.

File: configs/multilabel_classification/mobilenetv3_large.yml (hunk 2 above)
Comment (kprokofi) [reply configs/multilabel_classification/mobilenetv3_large.yml]: Yep, I forgot to return the batch size back

File: torchreid/data/datamanager.py (hunk 4 above)
Comment (kprokofi) [reply torchreid/data/datamanager.py]: Do you propose to include a condition and add a new parameter to the config file?

File: tools/main.py (hunk 1 above)
Comment (kprokofi) [reply tools/main.py:66]: In my case, these two lines of code work slowly. I need to wait up to a minute to collect env info. So, since it is not a very necessary thing, I propose to remove it. But there is something wrong with my environment, and on the cluster, it is working pretty fast. I could return it back

File: torchreid/data/datamanager.py (hunk 4 above)
Comment (sovrasov) [reply torchreid/data/datamanager.py]: Yes, let's add it. It would make sense for HPO. It expects batch size to be actually exactly the same as in config.

**Review by sovrasov — APPROVED** [review]
